tmva/tmva/inc/TMVA/Tools.h

```diff
@@ -283,9 +289,14 @@ namespace TMVA {
 
 template<typename T> void TMVA::Tools::ReadAttr( void* node, const char* attrname, T& value )
 {
-   TString val;
-   ReadAttr( node, attrname, val );
-   std::stringstream s(val.Data());
+   // read attribute from xml
+   const char *val = xmlengine().GetAttr(node, attrname);
+   if (val == 0) {
+      const char *nodename = xmlengine().GetNodeName(node);
+      Log() << kFATAL << "Trying to read non-existing attribute '" << attrname << "' from xml node '" << nodename << "'"
+            << Endl;
+   }
```

Comment:
i had assumed kFATAL would take care of that:) - (the previous author did) - shall I add an else?